### PR TITLE
Update to Lit v2

### LINKED ISSUE #201: Update to the new Lit v2
Current LitElement draft: https://github.com/Polymer/lit-html/pull/1231

Work in progress PR: https://github.com/IBM/pwa-lit-template/pull/202

### PR DESCRIPTION
**Pull request by abdonrd**
Fix #201 

~~Waiting for https://github.com/Polymer/lit-html/pull/1231~~

### CODE REVIEW

**Comment by abdonrd** [comment]
Blocked by https://github.com/modernweb-dev/web/issues/1376

**Comment by SinaKarimi7** [comment]
Well done @abdonrd, when will it be merged into `main` branch?

**Comment by abdonrd** [comment]
> Well done @abdonrd, when will it be merged into `main` branch?

We was waiting for Lit v2 final release.

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@ name: Deploy
 
 on:
   push:
-    branches: [main]
+    branches: [main, lit-next]
```
Comment (Tansito) [inline .github/workflows/deploy.yml:5]: Can we remove this change now that we are going to use Lit v2 in the main branch?

File: rollup.config.js [hunk 2]
```diff
@@ -40,6 +40,18 @@ const config = merge(
   createSpaConfig({
     outputDir: DIST_PATH,
     legacyBuild: true,
+    polyfillsLoader: {
```
Comment (Tansito) [inline rollup.config.js:43]: Probably I'm wrong here, @abdonrd but it's this polyfill needed now?

Comment (abdonrd) [reply rollup.config.js:43]: You can check here the Lit docs:
https://lit.dev/docs/releases/upgrade/#load-polyfill-support-when-using-web-components-polyfills

**Review by Tansito — APPROVED** [review]

**Pull request merged**